test/provision/docker-run-cilium-docker-plugin.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+CILIUM_DOCKER_PLUGIN_IMAGE=${CILIUM_DOCKER_PLUGIN_IMAGE:-cilium/docker-plugin:latest}
+
+set -e
+shopt -s extglob
+
+# Run without sudo if not available (e.g., running as root)
+SUDO=
+if [ ! "$(whoami)" = "root" ] ; then
```

Comment:
nit, `root` is just a name, but `uid=0` is always root, so maybe best to check `id` or the env variable `$EUID` if you want this script to run in pedantic environments